### PR TITLE
Cleanup: remove prometheus dependencies for e2e_node

### PR DESCRIPTION
**Pull request by gavinfish**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

remove prometheus dependencies for e2e_node

#### Which issue(s) this PR fixes:

Xref #89267

#### Does this PR introduce a user-facing change?

```release-note
none
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by RA489** [comment]
@gavinfish Please rebase this PR.

**Comment by gavinfish** [comment]
@RA489 This PR is pending on #99876, will update when that issue is resolved.

**Comment by BenTheElder** [comment]
Re: https://github.com/kubernetes/kubernetes/pull/99308#issuecomment-792702704

Sorry that took me so long to get to, there's been so much review lately 😬 It's done now.

This PR will also need to rebase & drop the BUILD file changes (see Kubernetes dev announcement for more details but basically hack/update-bazel.sh during rebase will handle it).

**Comment by gavinfish** [comment]
@BenTheElder Thanks 😄 !

@RainbowMango PTAL for metrics.

**Comment by RainbowMango** [comment]
/test pull-kubernetes-e2e-kind
cc @serathius

File: staging/src/k8s.io/component-base/metrics/testutil/metrics.go [hunk 1]
```diff
@@ -38,6 +39,22 @@ var (
 	QuantileLabel model.LabelName = model.QuantileLabel
 )
 
+// Sample is model.Sample.
+type Sample model.Sample
+
+// NewSample returns new Sample which is initialized.
+func NewSample(kv map[string]string, value float64, time int64) *Sample {
```
Comment (oomichi) [inline staging/src/k8s.io/component-base/metrics/testutil/metrics.go:46]: By checking callers of this function, `time` is always 0.
Do you have a plan to use this `time` for the other value?
If not, it is nice to remove it from this function for easy code.

File: staging/src/k8s.io/component-base/metrics/testutil/metrics.go [hunk 2]
```diff
@@ -68,6 +85,19 @@ func NewMetrics() Metrics {
 	return result
 }
 
+// NewMetricsFromSample returns new Metrics which is initialized with Sample.
+func NewMetricsFromSample(kv map[string][]*Sample) Metrics {
```
Comment (oomichi) [inline staging/src/k8s.io/component-base/metrics/testutil/metrics.go:89]: This function is called from test/e2e_node/dynamic_kubelet_config_test.go only.
Why don't we put this function in the file directly?
The following 2 functions also are called in the file only.

I might miss some point.

**Review by oomichi — COMMENTED** [review]
/cc @oomichi

**Comment by SergeyKanzhelev** [comment]
/lgtm cancel

@gavinfish I'm cancelling lgtm and moving this PR into "Waiting for author". Please address @oomichi 's recommendations

File: staging/src/k8s.io/component-base/metrics/testutil/metrics.go (hunk 1 above)
Comment (gavinfish) [reply staging/src/k8s.io/component-base/metrics/testutil/metrics.go]: done

File: staging/src/k8s.io/component-base/metrics/testutil/metrics.go (hunk 2 above)
Comment (gavinfish) [reply staging/src/k8s.io/component-base/metrics/testutil/metrics.go:88]: This PR is mainly to remove k/k dependency on prometheus. So move functions used prometheus to component-base/metrics.

**Comment by gavinfish** [comment]
@oomichi PTAL

File: staging/src/k8s.io/component-base/metrics/testutil/metrics.go (hunk 2 above)
Comment (SergeyKanzhelev) [reply staging/src/k8s.io/component-base/metrics/testutil/metrics.go:88]: is there plan to remove the reference from `component-base/metrics` later? Is the benefit of moving it here that all references will be addressed at once?

File: staging/src/k8s.io/component-base/metrics/testutil/metrics.go [hunk 3]
```diff
@@ -433,3 +471,11 @@ func LabelsMatch(metric *dto.Metric, labelFilter map[string]string) bool {
 
 	return true
 }
+
+// GetModelTimeFunc returns a function which gets the local Time corresponding to the given model time.
+func GetModelTimeFunc(sec int64) interface{} {
```
Comment (SergeyKanzhelev) [inline staging/src/k8s.io/component-base/metrics/testutil/metrics.go:476]: what would be the use case when `sec` will be passed here?

**Comment by dims** [comment]
Is this PR still needed, please rebase if so (or we can close it?)

**Comment by dims** [comment]
This PR is older than 4 weeks and needs as rebase.

Please rebase the PR against latest master and reopen if still needed.

/close

**Pull request closed without merging**